driver-core/src/main/com/mongodb/client/model/Aggregates.java

```diff
@@ -67,6 +69,23 @@ public static Bson addFields(final List<Field<?>> fields) {
         return new AddFieldsStage(fields);
     }
 
+    /**
+     * Creates an $addFields pipeline stage
+     *
+     * @param fields        the fields to add
+     * @param <TExpression> the expression type
+     * @return the $addFields pipeline stage
+     * @see com.mongodb.client.model.expressions.Expressions
+     * @mongodb.driver.manual reference/operator/aggregation/addFields/ $addFields
+     * @mongodb.server.release 3.4
+     * @since 4.?
+     */
+    public static <TExpression> Bson addFields(final Map<String, TExpression> fields) {
```

Comment:
This overload lets you use expressions more easily, where each value in the map is an expression, e.g.

    addFields(Map.of(
                        "foo", add(...),
                        "bar", switch(...)
                      ))

Reply:
Is order important?

Reply:
No, there's no reason I can think of that order should matter here, and in general it shouldn't be relied on as the server doesn't guarantee that order is preserved.